feat: builtin resource definitions

shanewxy opened the pull request:
<!-- IMPORTANT: Please do not create a Pull Request without creating an issue first. -->
**Problem:**
Add builtin resource definitions for kubernetes and docker.

**Related Issue:**
#1708

thxCode commented on pkg/resourcedefinitions/builtin.go:
```diff
@@ -0,0 +1,115 @@
+)
+
+var (
+	logger = log.WithName("builtin").WithName("resource-definitions")
```
keep the logger in function scope

thxCode commented on pkg/resourcedefinitions/builtin.go:
```diff
@@ -0,0 +1,115 @@
+
+	version, err := mc.TemplateVersions().Query().
+		Where(templateversion.Name(name)).
+		Order(model.Desc(templateversion.FieldVersion)).
```
we should compare the semver versions here?

thxCode commented on pkg/resourcedefinitions/builtin.go:
```diff
@@ -0,0 +1,115 @@
+var (
+	logger = log.WithName("builtin").WithName("resource-definitions")
+
+	resourceTypes  = []string{"containerservice", "mysql", "postgresql", "redis"}
+	connectorTypes = []string{types.ConnectorTypeKubernetes, types.ConnectorTypeDocker}
```
work with repo topics

gitlawr commented on pkg/resourcedefinitions/builtin.go:
```diff
@@ -0,0 +1,162 @@
+	}
+
+	if connectorType == types.ConnectorTypeDocker {
+		v, err := json.Marshal(map[string]string{"network_id": "local-walrus"})
```
Do you think we can use this as the default value in docker templates?

shanewxy replied:
Yes. I'll modify the templates schema and remove this.

gitlawr commented:
Can we provide a way to opt out of creating the builtin definitions? In case when users want to define their own `mysql`, `redis`, etc.

shanewxy commented:
> Can we provide a way to opt out of creating the builtin definitions? In case when users want to define their own `mysql`, `redis`, etc.

Updated. Added a setting for builtin resource definitions.

gitlawr reviewed (approved):
LGTM

thxCode reviewed (approved):
LGTM
